### PR TITLE
Feature/encode basic info without ssl

### LINKED ISSUE #1330: Encode selected information
**Is your feature request related to a problem? Please describe.**
In the http protocol environment, packets can be exposed as is. Although not normally required, adding the ability to encode and transmit selected information will help enhance security.

**Describe the solution you'd like**
Selective transferring with encrypted payload

### PR DESCRIPTION
**Pull request by inureyes**
This PR resolves #1330 

Also, it includes pure ESM-based written code without node.js crypto library.

### CODE REVIEW

File: src/lib/backend.ai-client-esm.ts [hunk 1]
```diff
@@ -0,0 +1,4082 @@
...
+   * Update user's password.
+   *
+   */
+  async update_password(oldPassword, newPassword, newPassword2) {
+    let body = {
+      'old_password': oldPassword,
+      'new_password': newPassword,
+      'new_password2': newPassword2
+    };
+    let rqst = this.newSignedRequest('POST', `/auth/update-password`, body);
```
Comment (adrysn) [inline src/lib/backend.ai-client-esm.ts]: This request should also be secure.

Comment (inureyes) [reply src/lib/backend.ai-client-esm.ts]: @adrysn fixed!

File: src/lib/backend.ai-client-node.js [hunk 2]
```diff
@@ -1126,6 +1127,15 @@ class Client {
         else {
             hdrs.set('Content-Type', content_type);
         }
+        // Add secure tag if payload is encoded.
+        if (false) {
+            console.log("test");
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.js]: is this console.log intended??

File: src/lib/backend.ai-client-node.js [hunk 3]
```diff
@@ -1191,6 +1201,21 @@ class Client {
         let t = year + month + day;
         return t;
     }
+    getEncodedPayload(body) {
+        const iv = crypto_node.randomBytes(16);
+        console.log(iv);
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.js]: same here

File: src/lib/backend.ai-client-node.ts [hunk 4]
```diff
@@ -1221,6 +1222,16 @@ class Client {
     } else {
       hdrs.set('Content-Type', content_type);
     }
+    // Add secure tag if payload is encoded.
+    if (false) {
+      console.log("test");
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.ts]: same here

File: src/lib/backend.ai-client-node.ts [hunk 5]
```diff
@@ -1289,6 +1300,27 @@ class Client {
     return t;
   }
 
+  getEncodedPayload(body) {
+    const iv = crypto_node.randomBytes(16);
+    console.log(iv);
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.ts]: same here

File: src/lib/backend.ai-client-node.js [hunk 6]
```diff
@@ -1126,6 +1127,15 @@ class Client {
         else {
             hdrs.set('Content-Type', content_type);
         }
+        // Add secure tag if payload is encoded.
+        if (false) {
+            console.log("test");
+            if (typeof requestBody == 'string') {
+                hdrs.set('X-BackendAI-Encoded', 'true');
+                console.log(typeof requestBody);
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.js]: same here

File: src/lib/backend.ai-client-node.ts [hunk 7]
```diff
@@ -1289,6 +1300,27 @@ class Client {
...
+    try {
+      const cipher = crypto_node.createCipheriv(
+        'aes-256-cbc',
+        Buffer.from('bf3c199c2470cb477d907b1e0917c17b'),
+        iv
+      );
+      console.log(cipher);
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.ts]: same here

File: src/lib/backend.ai-client-node.ts [hunk 8]
```diff
@@ -1289,6 +1300,27 @@ class Client {
...
+      const encrypted = cipher.update(body);
+      return (
+        iv.toString('hex') +
+        ':' +
+        Buffer.concat([encrypted, cipher.final()]).toString('hex')
+      )
+    } catch (error) {
+      console.error(error);
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.ts]: is this an error handling process?

**Review by fregataa — COMMENTED** [review]
there are some console.log statements. Im wondering they are intended

**Comment by inureyes** [comment]
> there are some console.log statements. Im wondering they are intended

Can you point out the location?

File: src/lib/backend.ai-client-node.js [hunk 9]
```diff
@@ -1191,6 +1201,21 @@ class Client {
         let t = year + month + day;
         return t;
     }
+    getEncodedPayload(body) {
+        const iv = crypto_node.randomBytes(16);
+        console.log(iv);
+        try {
+            const cipher = crypto_node.createCipheriv('aes-256-cbc', Buffer.from('bf3c199c2470cb477d907b1e0917c17b'), iv);
+            console.log(cipher);
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.js]: mark

File: src/lib/backend.ai-client-node.js [hunk 10]
```diff
@@ -1191,6 +1201,21 @@ class Client {
...
+            const encrypted = cipher.update(body);
+            return (iv.toString('hex') +
+                ':' +
+                Buffer.concat([encrypted, cipher.final()]).toString('hex'));
+        }
+        catch (error) {
+            console.error(error);
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.js]: mark

File: src/lib/backend.ai-client-node.ts [hunk 11]
```diff
@@ -1221,6 +1222,16 @@ class Client {
     } else {
       hdrs.set('Content-Type', content_type);
     }
+    // Add secure tag if payload is encoded.
+    if (false) {
+      console.log("test");
+      if (typeof requestBody == 'string') {
+        hdrs.set('X-BackendAI-Encoded', 'true');
+        console.log(typeof requestBody);
```
Comment (fregataa) [inline src/lib/backend.ai-client-node.ts]: mark

File: src/lib/backend.ai-client-node.js (hunk 2 above)
Comment (inureyes) [reply src/lib/backend.ai-client-node.js]: Ah; That js file will be disappeared... after merging.

**Review by adrysn — APPROVED** [review]
LGTM.

**Pull request merged**